### PR TITLE
[Core] Feature/#263 add SocketAuthenicator and JWT

### LINKED ISSUE #284: - JWT

### LINKED ISSUE #285: - JWT Signer

### LINKED ISSUE #286: - JWT Encoder

### LINKED ISSUE #287: - SocketAuthenticator

### PR DESCRIPTION
**Pull request by llbartekll**
Adds SocketAuthenicator, JWT, JWT signer for client authentication on Iridium with unit tests for dependant components.
close #284  
close #285
close #286
close  #287

### CODE REVIEW

File: Sources/WalletConnectRelay/ClientAuth/JWT/JWT.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+struct JWT: Codable, Equatable {
```
[the remark this answers is not in the document]
Comment (flypaper0) [reply Sources/WalletConnectRelay/ClientAuth/JWT/JWT.swift:3]: If so, no need to modify alg inside `Header`. We could just recreate `Header`

File: Sources/WalletConnectRelay/ClientAuth/JWT/JWTSigning.swift [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+protocol JWTSigning {
+    var alg: String {get}
+    func sign(header: String, claims: String) throws -> String
+}
+
+struct EdDSASigner: JWTSigning {
+    enum Errors: Error {
+        case invalidJWTString
+    }
+    var alg = "EdDSA"
```
Comment (flypaper0) [inline Sources/WalletConnectRelay/ClientAuth/JWT/JWTSigning.swift:13]: why alg everywhere is var?

File: Tests/RelayerTests/AuthTests/ED25519DIDKeyTests.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import Foundation
+import XCTest
+@testable import WalletConnectRelay
+
+final class ED25519DIDKeyFactoryTests: XCTestCase {
+    let expectedDid = "did:key:z6MkodHZwneVRShtaLf8JKYkxpDGp1vGZnpGmdBpX8M2exxH"
+    var sut: ED25519DIDKeyFactoryImpl!
+
+    func test() {
+
```
Comment (flypaper0) [inline Tests/RelayerTests/AuthTests/ED25519DIDKeyTests.swift:10]: planning to implement?

**Review by flypaper0 — APPROVED** [review]

File: Tests/RelayerTests/AuthTests/ED25519DIDKeyTests.swift (hunk 3 above)
Comment (llbartekll) [reply Tests/RelayerTests/AuthTests/ED25519DIDKeyTests.swift]: implemented but the factory itself is not implemented - scope of #288

**Pull request merged**